web/client/utils/ConfigUtils.js

```diff
@@ -50,6 +50,38 @@ const getConfigurationOptions = function(query, defaultName, extension, geoStore
     };
 };
 
+/**
+ * it removes some params from the query string
+ * return the shrinked url
+*/
+const removeParameters = (urlToFilter, skip) => {
+    const urlparts = urlToFilter.split('?');
+    let paramsFiltered = "";
+    if (urlparts.length >= 2 && urlparts[1]) {
+        const pars = urlparts[1].split(/[&;]/g).filter( p => !!p);
+        pars.forEach((par, i) => {
+            const param = par.split('=');
+            if (skip.indexOf(param[0].toLowerCase()) === -1) {
+                let addAnd = i === (pars.length - 1) ? "" : "&";
+                paramsFiltered += param.join("=") + addAnd;
+            }
+        });
+    }
+    return !!paramsFiltered ? urlparts[0] + "?" + paramsFiltered : urlparts[0];
+};
+/**
+ * WORKAROUND: it removes the extra ? when the authkey param is present
```

Comment:
Please specify that the problem of the extra ? is due to a configuratbility problem of the csw plugin of GeoServer.

I give some you some details to insert in the comment, even to open a JIRA on GeoServer in the future for this:

The problem happens when you have this in the `Record.properties` file of the csw folder in GeoServer, with csw plugin installed: 
```
references.scheme='OGC:WMS'
references.value=strConcat('${url.wms}?service=WMS&request=GetMap&layers=', prefixedName)
```
That is ok when you are not using an authkey, but If you have the authkey module installed and you get record with a proper authkey parameter the ${url.wms} URL will have also `?authkey=...` and so the final URL is something like: 
```
http://domain.org/geoserver/?autkey=abcdefghijklmnopqrstuvz1234567890?service=WMS&request=GetMap&layers=LAYER_NAME
```
 `${url.wms}` is replaced with the wms URL after the execution of `strConcat`, so the url can not be parsed in any way  to solve the problem via configuration, because you can not know if ${url.wms} contains `?` or not.